### PR TITLE
Add MessageMetadata type

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-message/src/output/signature_locked_single.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ impl Packable for SignatureLockedSingleOutput {
     }
 
     fn unpack<U: Unpacker>(unpacker: &mut U) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
-        let address = Address::unpack(unpacker).coerce()?;
+        let address = Address::unpack(unpacker)?;
```
Comment (Adam-Gleave) [inline bee-message/src/output/signature_locked_single.rs:60]: Every time I forget!

**Review by Adam-Gleave — DISMISSED** [review]
🤌

File: bee-message/src/metadata.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    /// The flags describing the message state.
+    flags: Flags,
+    /// The local time the message was received by the node.
+    received_timestamp: u64,
+    /// The local time the message got solid.
+    solidification_timestamp: u64,
+    /// The branch ID of the message, i.e. the part of the Tangle where the message is located.
+    branch_id: [u8; 32],
+    /// Contains the node's opinion on the issuing timestamp of a message.
+    opinion: Opinion,
```
Comment (samuel-rufi) [inline bee-message/src/metadata.rs:113]: opinion for ? should it be renamed to `FpcOpinion` ?

File: bee-message/src/metadata.rs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+pub struct MessageMetadata {
+    /// The flags describing the message state.
+    flags: Flags,
+    /// The local time the message was received by the node.
+    received_timestamp: u64,
+    /// The local time the message got solid.
+    solidification_timestamp: u64,
+    /// The branch ID of the message, i.e. the part of the Tangle where the message is located.
+    branch_id: [u8; 32],
+    /// Contains the node's opinion on the issuing timestamp of a message.
```
Comment (samuel-rufi) [inline bee-message/src/metadata.rs:112]: Is Fpc really voting on the issuing timestamp of a message?

**Review by samuel-rufi — CHANGES_REQUESTED** [review]

File: bee-message/src/metadata.rs (hunk 3 above)
Comment (thibault-martinez) [reply bee-message/src/metadata.rs:112]: Yes

Comment (thibault-martinez) [reply bee-message/src/metadata.rs:112]: https://github.com/iotaledger/IOTA-2.0-Research-Specifications/blob/main/2.2%20Message%20Layout.md#226-metadata

File: bee-message/src/metadata.rs (hunk 2 above)
Comment (thibault-martinez) [reply bee-message/src/metadata.rs:113]: I'd rather keep the names from the spec for consistency tbh

File: bee-message/src/metadata.rs (hunk 3 above)
Comment (samuel-rufi) [reply bee-message/src/metadata.rs:112]: okey interesting, thx!

File: bee-message/src/metadata.rs (hunk 2 above)
Comment (samuel-rufi) [reply bee-message/src/metadata.rs:113]: okey makes sense to go with the spec

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**